tasks/eva_3450/src/main/groovy/eva3450/CreateSplitCandidates.groovy

```diff
@@ -0,0 +1,121 @@
+import uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment
+
+import static org.springframework.data.mongodb.core.query.Criteria.where
+import static org.springframework.data.mongodb.core.query.Query.query
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.getDbsnpSveClass
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.getSveClass
+
+class CreateSplitCandidates {
+    static def logger = LoggerFactory.getLogger(CreateSplitCandidates.class)
+
+    String assembly
+    String splitCandidatesFile
+    EVADatabaseEnvironment prodEnv
+    EVADatabaseEnvironment devEnv
+    int batchSize
+
+    CreateSplitCandidates() {}
+
+    CreateSplitCandidates(String assembly, String splitCandidatesFile, EVADatabaseEnvironment prodEnv,
+                          EVADatabaseEnvironment devEnv, int batchSize) {
+        this.assembly = assembly
+        this.splitCandidatesFile = splitCandidatesFile
+        this.prodEnv = prodEnv
+        this.devEnv = devEnv
+        this.batchSize = batchSize
+    }
+
+
+    def process = {
+        def is = new File(splitCandidatesFile).newInputStream()
+        logger.info("Create Split candidate documents in batches of ${batchSize}")
+        int totalInsertedCount = 0
+        int lineRead = 0
+        def splitCandidateOperations = []
+        ArrayList<String> batchedLines = new ArrayList<String>()
+        is.eachLine{
+            lineRead = lineRead + 1
+            batchedLines.add(it)
+            if (batchedLines.size() == batchSize){
+                splitCandidateOperations = createBatchSplitCandidates(batchedLines)
+                totalInsertedCount = insertOperations(splitCandidateOperations, totalInsertedCount, lineRead)
+                batchedLines = new ArrayList()
+            }
+        }
+        if (batchedLines.size() > 0){
+            splitCandidateOperations = createBatchSplitCandidates(batchedLines)
+            totalInsertedCount = insertOperations(splitCandidateOperations, totalInsertedCount, lineRead)
+        }
+        is.close()
+
+    }
+
+    def insertOperations = {ArrayList splitCandidateOperations, int totalInsertedCount, int lineRead ->
+        def insertedCount = devEnv.bulkInsertIgnoreDuplicates(splitCandidateOperations, SubmittedVariantOperationEntity.class)
+        totalInsertedCount = totalInsertedCount + insertedCount
+        logger.info("Read ${lineRead}, Inserted ${totalInsertedCount} Split candidate documents")
+        return totalInsertedCount
+    }
+
+    def createBatchSplitCandidates = { List<String> rsidAndHashes ->
+        Map<Long, String[]> rsidToHashes = rsidAndHashes.collectEntries {
+            String[] sp_line = it.split("\\t")
+            def rsid = sp_line.first().toLong()
+            def hashes = sp_line.tail()
+            return [ (rsid): hashes]
+        }
+        def rsids = rsidToHashes.keySet().toList()
+        def evaAndDbsnpSveCursors = [sveClass, dbsnpSveClass].collect { collectionClass ->
```

Comment:
You can use `collectMany` to simplify this a bit - example courtesy of Nitin [here](https://github.com/EBIvariation/eva-tasks/blob/1b8abeea8e626ca248aedbee3438e1f7de3afe8d/tasks/eva_3417/src/main/groovy/eva3417/RemediateLowerCaseRefAlt.groovy#L87)